File: set_1/arrays/python/f35a7dbf-db98-4055-b459-8f8a1624845d.py
arr1 = [-1, 2, -3, 4, 5, 6, -7, 8, 9]

def pos_neg_rearrange1(arr):
    array_length = len(arr)
    i = -1
    for j in range(array_length):
        if (arr[j] < 0):
            i += 1
            # swapping of arr
            arr[i], arr[j] = arr[j], arr[i]

    pos_idx, neg_idx = i+1, 0

    while (pos_idx < array_length and neg_idx < pos_idx and arr[neg_idx] < 0):

        arr[neg_idx], arr[pos_idx] = arr[pos_idx], arr[neg_idx]
        pos_idx += 1
        neg_idx += 2
    return arr


resp1 = pos_neg_rearrange1(arr1)
print(f'The new rearranged array is --> {resp1}')

File: set_1/arrays/javascript/f35a7dbf-db98-4055-b459-8f8a1624845d.js
const arr1 = [-9, -92, -21, -90, 51, -73, -30, -66, -22, 48];

const arr2 = [5, -3, 37, -40, 57, 52, -99, -11, -57, 28];

const posNegRearrange1 = (arr, n) => {
    const arrayLength = arr.length;
    let i = -1, temp = 0;
    for (let j = 0; j < arrayLength; j++) {
        if (arr[j] < 0) {
            i++;
            temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
    let posIdx = i + 1, negIdx = 0;
    while (posIdx < arrayLength && negIdx < posIdx && arr[negIdx] < 0) {
        temp = arr[negIdx];
        arr[negIdx] = arr[posIdx];
        arr[posIdx] = temp;
        posIdx++;
        negIdx += 2;
    }
    return arr;
}


const resp1 = posNegRearrange1(arr1);
console.log(`The new reaggranged array is --> ${resp1}`);

const resp2 = posNegRearrange1(arr2);
console.log(`The new reaggranged array is --> ${resp2}`);